### PR TITLE
Add security information (IP and browser) to the password reset email

### LINKED ISSUE #4345: Improve lost password email
Followup to #4308

Mockup by @Espina2 

new password

Comments by @eppfel:

> Yeah, @Espina2 copy would be much appreciated. 
> 
> Too minor edits:
> 
> This ~~reset~~ password _reset_ is only available... 
> 
> Or:
> 
> This password reset link is only.... 
> 
> And 
> 
> ~~For security,~~ this request...

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: #4345

## Summary
Add some metadata like the IP address and the browser type to the password reset mail for fraud detection.

## Screenshots

before | after
---|---
image | image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Screenshots before/after for front-end changes

### CODE REVIEW

File: core/Controller/LostController.php [hunk 1]
```diff
@@ -308,12 +308,31 @@ protected function sendEmail(string $input): void {
...
+			$platform = 'Opera';
+		} elseif ($this->request->isUserAgent(['/Edg.*\//'])) {
+			$platform = 'Edge';
+		} elseif ($this->request->isUserAgent(['/Chrome\//'])) {
+			$platform = 'Chrome';
+		} else {
+			$platform = $this->l10n->t('unknown');
+		}
+		$emailTemplate->addBodyText(
+			htmlspecialchars($this->l10n->t('Security notice: This password reset was requested from following IP address and browser: %1$s (%2$s).', [ $this->request->getRemoteAddress(), $platform ]))
```
Comment (nickvergessen) [inline core/Controller/LostController.php:333]: I would rephrase and reorder this information

> ## Password reset
>
> A password resset was requested from %1$s (IP address) and using %2$s.
> If you have not requested the password reset, then ignore this email.
> Otherwise, click the following link to reset your password.

File: core/Controller/LostController.php [hunk 2]
```diff
@@ -308,12 +308,31 @@ protected function sendEmail(string $input): void {
...
+		} elseif ($this->request->isUserAgent(['/Safari\//'])) {
+			$platform = 'Safari';
+		} elseif ($this->request->isUserAgent(['/(OPR|Opera)\//'])) {
+			$platform = 'Opera';
+		} elseif ($this->request->isUserAgent(['/Edg.*\//'])) {
+			$platform = 'Edge';
+		} elseif ($this->request->isUserAgent(['/Chrome\//'])) {
+			$platform = 'Chrome';
+		} else {
+			$platform = $this->l10n->t('unknown');
```
Comment (nickvergessen) [inline core/Controller/LostController.php:330]: we should make sure that it does not show "unknown" when reset is done via a mobile app or the desktop client.

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: core/Controller/LostController.php (hunk 2 above)
Comment (susnux) [reply core/Controller/LostController.php:336]: Good point, fixed.

File: core/Controller/LostController.php (hunk 1 above)
Comment (susnux) [reply core/Controller/LostController.php]: I have added that additional paragraph because of this comment: https://github.com/nextcloud/server/issues/4345#issuecomment-296418122

File: core/Controller/LostController.php (hunk 2 above)
Comment (nickvergessen) [reply core/Controller/LostController.php:336]: Can you add the Talk clients too? 😅

File: core/Controller/LostController.php (hunk 1 above)
Comment (nickvergessen) [reply core/Controller/LostController.php]: So I checked all reset password emails in my inbox and most of them did not have information like this.
But if they did, it was the first information given.

cc @jancborchard for your 6y old comment

Comment (nickvergessen) [reply core/Controller/LostController.php]: So I checked all reset password emails in my inbox and most of them did not have information like this.
But if they did, it was the first information given.

cc @jancborchardt for your 6y old comment

**Comment by come-nc** [comment]
Stupid question maybe, but is there no data privacy implications here?
This is sending the IP and browser of a web visitor to another person by email. I understand in this case it’s the web visitor’s fault for entering the email address of someone else but laws may not be that precise?

**Comment by tobiasKaminsky** [comment]
It is. Server is processing ip address and thus user must be informed prior to it.
(other services, e.g. Google go even further and try to locate ip address, to give user a better info where the login attempt was done, e.g "user tried to log in near Frankfurt")

As far as I know we do not ship by default such an information, so I tend to not include this at all.

File: core/Controller/LostController.php (hunk 1 above)
Comment (skjnldsv) [reply core/Controller/LostController.php]: also cc @nimishavijay :)

Comment (nickvergessen) [reply core/Controller/LostController.php]: As per GDPR officer we can not do this:
https://github.com/nextcloud/server/pull/37808#issuecomment-1515931164

Comment (skjnldsv) [reply core/Controller/LostController.php]: Then closing :)

**Comment by skjnldsv** [comment]
As per GDPR officer we can not do this:
https://github.com/nextcloud/server/pull/37808#issuecomment-1515931164

**Comment by susnux** [comment]
Would it make sense to only include the browser?
